File: average/AverageCalculatorFunctional.ts
const numbers: number[] = [1, 2, 3, 4, 5];

let sum: number = 0;
for (let i = 0; i < numbers.length; i++) {
  sum += numbers[i];
}
const average: number = sum / numbers.length;

console.log("Average:", average);

File: average/AverageCalculatorFunctional.java
import java.util.Arrays;
import java.util.List;

public class AverageCalculatorFunctional {
    public static void main(String[] args) {
        List<Integer> numbers = Arrays.asList(1, 2, 3, 4, 5);

        double average = numbers.stream()
                .mapToInt(Integer::intValue)
                .average()
                .orElse(Double.NaN);

        System.out.println("Average: " + average);
    }
}
